src/dune_pkg/local_package.ml

```diff
@@ -145,7 +145,27 @@ let of_package (t : Dune_lang.Package.t) =
   let name = Package.name t in
   match Package.original_opam_file t with
   | None ->
-    let dependencies = t |> Package.depends |> Dependency_formula.of_dependencies in
+    let depends =
+      match Package.depends t, Package.has_opam_file t with
+      | [], Exists true ->
+        let name = t |> Package.name |> Package_name.to_string in
+        let hints =
+          [ Pp.textf "Delete the package %S from `dune-project`" name
+          ; Pp.textf "Add a `depends` field in the `package` stanza of package %S" name
+          ]
+        in
+        User_warning.emit
+          ~hints
+          [ Pp.textf
+              "Package %s loaded from `dune-project` but does not declare any \
```

Comment:
I'd prefer to use %S as you did in the hint above in case it's not clear that the word following "Package" in the message is a package name.